src/replicator.rs

```diff
@@ -472,21 +558,145 @@ pub extern "C" fn c_property_decryptor(
+
+        let mut result = FLSliceResult_New(0);
+        if let Some(input) = input.to_vec() {
+            result = (*repl_conf_context)
+                .collection_property_decryptor
+                .map(|callback| {
+                    callback(
+                        scope.to_string(),
+                        collection.to_string(),
+                        document_id.to_string(),
+                        Dict::wrap(properties, &properties),
+                        key_path.to_string(),
+                        input.to_vec(),
+                        algorithm.to_string(),
+                        kid.to_string(),
+                        &error,
+                    )
+                })
+                .map_or(FLSliceResult_New(0), |v| match v {
+                    Ok(v) => FLSlice_Copy(from_bytes(&v[..]).get_ref()),
+                    Err(err) => {
+                        match err {
+                            EncryptionError::Temporary => {
+                                error!("Decryption callback returned with transient error");
+                                error = Error {
+                                    code: ErrorCode::WebSocket(503),
+                                    internal_info: None,
+                                };
+                            }
+                            EncryptionError::Permanent => {
+                                error!("Decryption callback returned with non transient error");
+                                error = Error::cbl_error(CouchbaseLiteError::Crypto);
+                            }
+                        }
+
+                        FLSliceResult::null()
+                    }
+                });
+        } else {
+            error!("Decryption input is None");
+            error = Error::cbl_error(CouchbaseLiteError::Crypto);
+        }
+
+        if error != Error::default() {
+            *cbl_error = error.as_cbl_error();
+        }
+        result
+    }
+}
+
 #[derive(Default)]
 pub struct ReplicationConfigurationContext {
+    #[deprecated(note = "please use `collection.push_filter` on default collection instead")]
     pub push_filter: Option<ReplicationFilter>,
+    #[deprecated(note = "please use `collection.pull_filter` on default collection instead")]
     pub pull_filter: Option<ReplicationFilter>,
+    #[deprecated(note = "please use `collection.conflict_resolver` on default collection instead")]
     pub conflict_resolver: Option<ConflictResolver>,
-    pub property_encryptor: Option<PropertyEncryptor>,
-    pub property_decryptor: Option<PropertyDecryptor>,
+    #[deprecated(
+        note = "please use `collection_property_encryptor` on default collection instead"
+    )]
+    pub default_collection_property_encryptor: Option<DefaultCollectionPropertyEncryptor>,
+    #[deprecated(
+        note = "please use `collection_property_decryptor` on default collection instead"
+    )]
```

Comment:
You deprecated some fields but you don't use them accordingly in the engine side ^^